src/Simplex_tree/concept/FiltrationValue.h

```diff
@@ -5,19 +5,65 @@
  *    Copyright (C) 2014 Inria
  *
  *    Modification(s):
+ *      - 2024/08 Hannah Schreiber: Update of the concept after several additions to the Simplex tree.
  *      - YYYY/MM Author: Description of the modification
  */
 
 /** \brief Value type for a filtration function on a cell complex.
-  *
-  * A <EM>filtration</EM> of a cell complex (see FilteredComplex) is
-  * a function \f$f:\mathbf{K} \rightarrow \mathbb{R}\f$ satisfying \f$f(\tau)\leq 
-  * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices 
-  * by increasing filtration values (breaking ties so as a simplex appears after 
-  * its subsimplices of same filtration value) provides an indexing scheme 
-  * (see IndexingTag).
-  */
-  struct FiltrationValue {
-    /** \brief Operator < is a StrictWeakOrdering. */
-    bool operator<(FiltrationValue f1, FiltrationValue f2);
-  };
+ *
+ * Needs to implement `std::numeric_limits<FiltrationValue>::has_infinity` and when it returns `true`,
+ * has to implement `std::numeric_limits<FiltrationValue>::infinity()`. If it returns `false`, has to
+ * implement `std::numeric_limits<FiltrationValue>::max()` instead.
+ *
+ * A <EM>filtration</EM> of a cell complex (see FilteredComplex) is
+ * a function \f$f:\mathbf{K} \rightarrow \mathbb{R}\f$ satisfying \f$f(\tau)\leq
+ * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices
+ * by increasing filtration values (breaking ties so as a simplex appears after
+ * its subsimplices of same filtration value) provides an indexing scheme
+ * (see IndexingTag).
+ */
+struct FiltrationValue {
+  /**
+   * @brief Has to construct the default value of FiltrationValue. E.g., 0 for a numerical value or {} for a vector.
+   */
+  FiltrationValue(0);
```

Comment:
```suggestion
  FiltrationValue();
```
?

Reply:
Well, I wanted to tell here that if the constructor is called with `0` as argument, we should get the desired default value for the filtration value. This is because `double()` or `int()` are not guaranteed to give us `0`, so I had to replace the default arguments `= 0` with `= Filtration_value(0)` instead of `= Filtration_value()`. Do you know a better way to do this?

Reply:
> This is because `double()` or `int()` are not guaranteed to give us `0`

Where did you read that?

Reply:
I am pretty sure I saw that at several places. But perhaps it is not true anymore? I remember `int i;` giving me quite some random values for `i` at some point. But this is a few years ago and perhaps it was C, not C++ ?

Reply:
`int i;` does not initialize `i`, so if you try to read it, you can get anything. But `int()` has always zero-initialized, as far as I can tell.

Reply:
Oh, ok, good to know that there is a difference! Then, I can remove it.